### PR TITLE
#163519146 Add comment's feature

### PR DESCRIPTION
**Pull request by oesukam**
#### What does this PR do?
Add CRUD API end-points for comments

#### Description of Task to be completed?
Have the following end-point working:
`POST api/v1/articles/:articleSlug/comments`
`GET api/v1/articles/:articleSlug/comments?page=number`
`PUT api/v1/comments/:commentId`
`DELETE api/v1/comments/:commentId`

#### How should this be manually tested?

- Download/clone the repo

- run `npm install`

- run `npm run migrate:seed`

- run `npm run test`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?
#163519146

#### Screenshots (if appropriate)
N/A

#### Questions:
N/A

### CODE REVIEW

File: __tests__/routes/article.test.js [hunk 1]
```diff
@@ -79,12 +79,13 @@ describe('articles', () => {
   });
 
   test('should return article', async () => {
-    expect.assertions(5);
+    expect.assertions(6);
     const res = await request(app).get(`${urlPrefix}/articles/${newArticle.slug}`);
     expect(res.status).toBe(200);
     expect(res.body).toBeDefined();
```
Comment (Nziranziza) [inline __tests__/routes/article.test.js]: not necessary

Comment (Nziranziza) [inline __tests__/routes/article.test.js]: This is not really necessary

File: __tests__/routes/article.test.js [hunk 2]
```diff
@@ -117,6 +118,16 @@ describe('articles', () => {
     expect(res.body.articlesCount).toBeDefined();
   });
 
+  test('Fetch Articles - should return articles by favorited tag test', async () => {
+    expect.assertions(5);
+    const res = await request(app).get(`${urlPrefix}/articles?favorited=${loginUser1.username}`);
+    expect(res.status).toBe(200);
+    expect(res.body).toBeDefined();
```
Comment (Nziranziza) [inline __tests__/routes/article.test.js]: not necessary

File: __tests__/routes/article.test.js (hunk 1 above)
Comment (oesukam) [reply __tests__/routes/article.test.js]: @Nziranziza It came from the previous work but I am going to remove it. Thank you for the feedback as we shall always refactor some part of the codebase when necessary

File: __tests__/routes/comment.test.js [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+  });
+
+  test("VIEW - should return articles' comments for page 1", async () => {
+    expect.assertions(5);
+    const res = await request(app)
+      .get(`${urlPrefix}/articles/${newArticle.slug}/comments?page=1`)
+      .set('Authorization', loginUser1.token);
+
+    expect(res.status).toBe(200);
+    expect(res.body).toBeDefined();
```
Comment (Nziranziza) [inline __tests__/routes/comment.test.js:175]: not necessary

**Review by Nziranziza — APPROVED** [review]

**Review by mcaleb808 — APPROVED** [review]

**Comment by oesukam** [comment]
@filschristian Can you try to run the project because it has a lot of Eslint complaints and I can not run it?

**Pull request merged**